src/main/scala/castalia/Delay.scala

```diff
@@ -19,9 +20,143 @@ trait Delay {
 
 
 trait DelayedDistribution extends Delay {
-  def distributedFuture[T](mean: Double,
-                           stdev: Double,
-                           dist: Distribution[Double]
-                          ): Distribution[Double] =
-    dist.map(_*stdev + mean)
-}
\ No newline at end of file
+  def normalDistribution(mean: Double,
+                            stdev: Double,
+                            dist: Distribution[Double]
+                           ): Distribution[Double] =
+    dist.map(_ * stdev + mean)
+
+  def gammaDistribution(k: Double, theta: Double): Distribution[Double] = {
+    Distribution.gamma(k, theta)
+  }
+
+  /**
+    * continuous probability distributions defined on the interval [0, 1]
+    * parametrized by two positive shape parameters, denoted by α and β,
+    * that appear as exponents of the random variable and control the shape of the distribution.
+    *
+    * @param mu mean
+    * @param sigma stdev
+    * @return Distribution obeying beta: for use scaling should be applied
+    */
+  def betaDistribution(mu: Double, sigma: Double): Distribution[Double] = {
+    val (a, b) = getBetaParametersFromMeanAndSigma(mu, sigma)
+    Distribution.beta(a, b)
+  }
+
+  def getBetaParametersFromMeanAndSigma(mu: Double, sigma: Double): (Double, Double) = {
+    val a = ((1 - mu)/math.pow(sigma,2)-(1/mu))* math.pow(mu,2)
+    val b = a * ((1/mu) - 1)
+    (a, b)
+  }
+
+  /**
+    *
+    * @param p95 the value to satisfy p95
+    * @param p99 the value to satisfy p99
+    * @return the WeibullDistribution
+    */
+  def getWeibullParametersFromPercentiles(p95: Double, p99: Double): (Double, Double) = {
+
+    def getGammaFromPercentiles: Double = {
+      (math.log(-math.log(1 - 0.99)) - math.log(-math.log(1 - 0.95))) / (math.log(p99) - math.log(p95))
+    }
+
+    def getBetaFromGamma(gamma: Double): Double = {
+      p95 / math.pow(-math.log(1 - 0.95), 1 / gamma)
+    }
+
+    val gamma = getGammaFromPercentiles
+    val beta = getBetaFromGamma(gamma)
+    (gamma, beta)
+  }
+
+  /**
+    * Generate weibull distribution
+    *
+    * @param p95
+    * @param p99
+    * @return
+    */
+  def weibullDistribution(p95: Double, p99: Double): Distribution[Double] = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/scala-academy/Castalia/file/1993066085/issues/source?bid=2998213&fileBranchId=3112713#l82)